src/redux/reducers.js

```diff
@@ -673,6 +787,20 @@ export function posts(state = {}, action) {
     case response(ActionTypes.GET_SINGLE_POST): {
       return updatePostData(state, action)
     }
+    case ActionTypes.REALTIME_POST_NEW:
+    case ActionTypes.REALTIME_POST_UPDATE: {
+      return { ...state, [action.post.id]: postParser(action.post) }
+    }
+    case ActionTypes.REALTIME_COMMENT_NEW: {
+      const post = state[action.comment.postId]
+      if (!post || post.comments && post.comments.indexOf(action.comment.id)!==-1) {
```

Comment:
For me, this condition is virtually unreadable. I would prefer readability and verbosity to this FP style one-liners.